Handling unsigned integers with bitshift unsigned operator

AlexPerrot opened the pull request:
Change embind fromWireType function for integers in order to handle unsigned values return as signed values by js.
Corrects Issue #3246.

kripken commented on src/embind/embind.js:
```diff
@@ -523,11 +523,15 @@ var LibraryEmbind = {
         name: name,
         'fromWireType': function(value) {
-            return value;
+            if(minRange !== 0) {
```
space after `if`

kripken commented on src/embind/embind.js:
```diff
@@ -523,11 +523,15 @@ var LibraryEmbind = {
+            if(minRange !== 0) {
+                return value;
+            }
+            return (value << bitshift)>>> bitshift;
```
space after `)`

kripken commented:
cc @chadaustin

chadaustin commented:
This needs a few unit tests in embind.test.js and embind_test.cpp. I wonder why this started happening: I'm pretty sure it did the right thing when I wrote embind at first. I wonder if there is a compiler setting or maybe fastcomp changed how embind integrates a bit.

Either way, I should have written comprehensive unit tests for numeric conversions, so if you can add a few that reproduce the but we will never have this problem again. :)

chadaustin commented:
Also, it's better to do the conditions outside the fromWireType functions and select an optimized fromWireType function.

AlexPerrot commented:
I moved the condition and added tests storing to memory and reloading the value.
Those tests fail without my modification to fromWireType and pass with it.

chadaustin commented:
Thanks! If it passes tests and all, looks good to me. @kripken can you merge this one?

kripken commented:
@AlexPerrot , please squash the commits into 1.

@chadaustin , I can merge it after that, but didn't I give you repo access so you could so such merges? If not I can add you now.

kripken commented:
Note that github doens't send update emails for commits, just comments - i might miss stuff like this in the future if you don't also comment.

chadaustin commented:
Thanks, @AlexPerrot.  Good catch and nice patch.
